OpenRA.Mods.Common/Traits/Render/RenderDebugState.cs

```diff
@@ -65,7 +65,7 @@ public void OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 
 		Color GetColor()
 		{
-			return self.EffectiveOwner != null ? self.EffectiveOwner.Owner.Color.RGB : self.Owner.Color.RGB;
+			return self.EffectiveOwner != null && self.IsInWorld ? self.EffectiveOwner.Owner.Color.RGB : self.Owner.Color.RGB;
```

Comment:
This heisenbug made a re-appearance in https://github.com/OpenRA/OpenRA/issues/11996 as well.